rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -2688,18 +2690,167 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 		return _spu->get_ch_value(ch);
 	}
 
-	void RDCH(spu_opcode_t op) //
+	static s64 exec_read_in_mbox(SPUThread* _spu)
 	{
-		update_pc();
-		value_t<s64> res;
-		res.value = call(&exec_rdch, m_thread, m_ir->getInt32(op.ra));
-		const auto next = llvm::BasicBlock::Create(m_context, "", m_function);
+		// TODO
+		return _spu->get_ch_value(SPU_RdInMbox);
+	}
+
+	static u32 exec_read_dec(SPUThread* _spu)
+	{
+		const u32 res = _spu->ch_dec_value - static_cast<u32>(get_timebased_time() - _spu->ch_dec_start_timestamp);
+
+		if (res > 1500)
+			std::this_thread::yield();
```

Comment:
need to check if spu loop detection is on before `std::this_thread::yield();`

Reply:
Well, yeah, I forgot